webapp/crashstats/libbugzilla.py

```diff
@@ -80,40 +106,55 @@ def crash_report_to_description(crash_report_url, processed_crash):
+        if threads:
+            if processed_crash.get("crashing_thread") is None:
+                lines.append("")
+                lines.append("No crashing thread identified; using thread 0.")
+
+            thread_index = processed_crash.get("crashing_thread") or 0
+            frames = minidump_thread_to_frames(threads[thread_index])
+
+    if not frames and "java_exception" in processed_crash:
+        stacktraces = (
+            processed_crash["java_exception"].get("exception", {}).get("values", [])
+        )
+        if stacktraces:
+            frames = java_exception_to_frames(
+                stacktraces[0]["stacktrace"].get("frames", [])
+            )
+
+    if frames:
         lines.append("")
-        lines.append("Java stack trace:")
+        if len(frames) == 1:
+            lines.append(f"Top {len(frames)} frame:")
+        else:
+            lines.append(f"Top {len(frames)} frames:")
         lines.append("```")
-        lines.append(truncate(processed_crash["java_stack_trace"], 5000))
+        for frame in frames:
+            signature = truncate(frame["signature"], 80)
+            lines.append(
+                f"{frame['frame']:<2} {frame['module']}  {signature}  {frame['source']}"
+            )
         lines.append("```")
 
-    # Generate frames from the stackwalker output from parsing a minidump
-    frames = None
-    threads = processed_crash.get("json_dump", {}).get("threads")
-    if threads:
-        if processed_crash.get("crashing_thread") is None:
-            lines.append("")
-            lines.append("No crashing thread identified; using thread 0.")
-
-        thread_index = processed_crash.get("crashing_thread") or 0
-        frames = bugzilla_thread_frames(threads[thread_index])
-
-        if frames:
-            lines.append("")
-            lines.append(f"Top {len(frames)} frames of crashing thread:")
-            lines.append("```")
-            for frame in frames:
-                signature = truncate(frame["signature"], 80)
-                lines.append(
-                    f"{frame['frame']:<2} {frame['module']}  {signature}  {frame['source']}"
-                )
-            lines.append("```")
+    elif processed_crash.get("java_stack_trace"):
+        # If there are no frames from a parsed minidump or a java_except value, then use
+        # the java_stack_trace if that exists; this is a big string blob, so we convert
+        # tabs to spaces, truncate it at 5,000 characters but don't otherwise format it
+        java_stack_trace = processed_crash["java_stack_trace"].replace("\t", "    ")
+        lines.append("")
+        lines.append("Java stack trace:")
+        lines.append("```")
+        lines.append(truncate(java_stack_trace, 5000))
+        lines.append("```")
 
-        else:
-            lines.append("")
-            lines.append("No stack.")
+    else:
+        lines.append("")
+        lines.append("No stack.")
```

Comment:
I changed things around so that if there isn't anything, then it now adds `No stack.`. Previously, it would have nothing which is weird and it's unclear what happened.